### PR TITLE
Switch the base image to debian stretch slim

### PR DESCRIPTION
**Pull request by roberthbailey**
**What this PR does / why we need it**: Changes the base image for the cluster api repo to debian.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
The base layer for cluster api containers changed from alpine to debian.
```

@kubernetes/kube-deploy-reviewers

### CODE REVIEW

File: cloud/terraform/cmd/terraform-machine-controller/Makefile [hunk 1]
```diff
@@ -18,7 +18,7 @@ GCR_BUCKET = k8s-cluster-api
 PREFIX = gcr.io/$(GCR_BUCKET)
 DEV_PREFIX ?= gcr.io/$(shell gcloud config get-value project)
 NAME = terraform-machine-controller
-TAG = 0.0.3
+TAG = 0.0.5
```
Comment (roberthbailey) [inline cloud/terraform/cmd/terraform-machine-controller/Makefile]: @krousey - I bumped this from 0.0.3 --> 0.0.5 to avoid conflicts w/ your in-flight PR.

**Comment by roberthbailey** [comment]
I haven't pushed new images to the cluster-api gcr buckets yet, but I will need to before this merges. If it looks sane, I'll go ahead and do the pushes.

**Comment by roberthbailey** [comment]
I'd added the rm to the end of each apt-get line and re-pushed my commit.

**Comment by tamalsaha** [comment]
Just curious, why the switch ?

**Comment by roberthbailey** [comment]
I've pushed new images for the 4 containers.

**Comment by vijaykatam** [comment]
> Just curious, why the switch ?

This comment was helpful for me https://github.com/kubernetes/kubernetes/issues/40248#issuecomment-280781931

**Pull request merged**